webapp/src/main/java/com/box/l10n/mojito/service/smartling/ThirdPartyTextUnitMatchingService.java

```
@@ -0,0 +1,201 @@
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static com.box.l10n.mojito.utils.Predicates.logIfFalse;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Service to manage matching between mojito text units and third party text units
+ **/
+@Service
+public class ThirdPartyTextUnitMatchingService {
+    /**
+     * logger
+     */
+    private static Logger logger = getLogger(ThirdPartyTextUnitMatchingService.class);
+
+    @Autowired
+    ThirdPartyTextUnitRepository thirdPartyTextUnitRepository;
+
+    @Autowired
+    SmartlingClient smartlingClient;
+
+    @Autowired
+    ThirdPartyTextUnitSearchService thirdPartyTextUnitSearchService;
+
+    String delimiter = "#@#";
+
+    private String fileRegex = ".*\\d+_(singular|plural)_source\\.xml";
+
+    private String pluralFileIndicator = "plural";
+
+    void getSmartlingThirdPartyTextUnits(String projectId) {
+        FilesResponse files = smartlingClient.getFiles(projectId);
+        for (File file : files.getResponse().getData().getItems()) {
+            String fileUri = file.getFileUri();
+            if (fileUri.matches(fileRegex)) {
+                processFile(fileUri, projectId);
+            }
+        }
+    }
+
+    public void processFile(String fileUri, String projectId) {
+        int maxResults = 500;
+
+        // pull all strings for a file then match
+        PageFetcherSplitIterator<StringInfo> sourceStringsPageFetcherSplitIterator = new PageFetcherSplitIterator<>(
+            (offset, limit) -> {
+                List<StringInfo> stringInfoList = new ArrayList<>();
+                SourceStringsResponse sourceStringsResponse = smartlingClient.getSourceStrings(projectId, fileUri, offset);
+                if (sourceStringsResponse.isSuccessResponse()) {
+                    stringInfoList = sourceStringsResponse.getResponse().getData().getItems();
+                }
```

this will ignore errors and process nothing?

---

If its not a success response, there will be no data to process but I'll add a log message

---

and added a test